### PR TITLE
Offline transformations exposed to python with pybind11

### PR DESCRIPTION
**Pull request by tomdol**
### Details:
 - Offline transformations transformation from cython to pybind11
 - The old code is left in the original location but not built any more (changes in inference-engine/ie_bridges/python/CMakeLists.txt)
 - The new bindings accept the `ov::Function` object instead of `IENetwork` - this has been reflected in the tests and at the call site in MO

### Tickets:
 - 61868

### CODE REVIEW

File: runtime/bindings/python/src/pyopenvino/core/ie_offline_transformations.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+//
+
+#include "pyopenvino/core/ie_offline_transformations.hpp"
+
+#include <generate_mapping_file.hpp>
+#include <moc_transformations.hpp>
+#include <pot_transformations.hpp>
+#include <pruning.hpp>
+
+// #include "ie_api_impl.hpp"
```
Comment (jiwaszki) [inline runtime/bindings/python/src/pyopenvino/core/ie_offline_transformations.cpp:12]: Is there a reason to keep this comment?

File: inference-engine/ie_bridges/python/CMakeLists.txt [hunk 2]
```diff
@@ -64,7 +64,6 @@ endfunction()
 
 set (PYTHON_BRIDGE_SRC_ROOT ${CMAKE_CURRENT_SOURCE_DIR})
 add_subdirectory (src/openvino/inference_engine)
-add_subdirectory (src/openvino/offline_transformations)
```
Comment (jiwaszki) [inline inference-engine/ie_bridges/python/CMakeLists.txt:67]: We should not remove it from old API.

File: inference-engine/ie_bridges/python/wheel/CMakeLists.txt [hunk 3]
```diff
@@ -34,7 +34,7 @@ endif()
 
 configure_file(${SETUP_ENV} "${CMAKE_CURRENT_SOURCE_DIR}/.env" @ONLY)
 
-add_custom_target(ie_wheel ALL DEPENDS ie_api offline_transformations_api)
```
Comment (jiwaszki) [inline inference-engine/ie_bridges/python/wheel/CMakeLists.txt:37]: Not to remove.

File: model-optimizer/mo/back/offline_transformations.py [hunk 4]
```diff
@@ -40,13 +40,15 @@ def apply_offline_transformations(input_model: str, framework: str, transforms:
 
     from openvino.inference_engine import read_network  # pylint: disable=import-error,no-name-in-module
     from openvino.offline_transformations import GenerateMappingFile  # pylint: disable=import-error,no-name-in-module
+    from ngraph import function_from_cnn # pylint: disable=import-error,no-name-in-module
 
     net = read_network(input_model + "_tmp.xml", input_model + "_tmp.bin")
-    apply_user_transformations(net, transforms)
-    apply_moc_transformations(net)
+    function = function_from_cnn(net)
```
Comment (jiwaszki) [inline model-optimizer/mo/back/offline_transformations.py:46]: To be deprecated.

File: inference-engine/ie_bridges/python/tests/test_offline_api.py [hunk 5]
```diff
@@ -1,52 +1,52 @@
 # Copyright (C) 2018-2021 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
-from openvino.inference_engine import IECore, IENetwork
-from openvino.offline_transformations import ApplyMOCTransformations, ApplyLowLatencyTransformation, ApplyPruningTransformation
+from openvino.offline_transformations import ApplyMOCTransformations, ApplyPOTTransformations, \
```
Comment (akuporos) [inline inference-engine/ie_bridges/python/tests/test_offline_api.py:4]: Please, move this test file to runtime/bindings/python/tests

**Pull request closed without merging**